### DISCUSSION TITLE
How to close/kill the app after every test in flutter integration test?

### DISCUSSION THREAD

**Discussion by jagadeeshpurplle** [appium/appium#17794] [Q&A]
Hi Team,

I'm using flutter integration driver to write integration tests for my mobile flutter app.

Below is the sample code of writing multiple `testWidgets` in a single `app_test.dart` file, after completion of first test execution i see the app directly opening from home screen and not from fresh app install. `Is this expected?`

I'm expecting it to open from fresh state where i can login again & proceed further.

Is there way to uninstall the app or kill the app after executing the first test case, so that my test execution always starts from fresh app open.

```
void main() {
  IntegrationTestWidgetsFlutterBinding.ensureInitialized();

  group('regression test suite', () {

    tearDown(() async {
      final getIt = GetIt.instance;
      await getIt.reset();
    });

    testWidgets('sample test', (tester) async {
      app.main();
      SLogger.i('App opened');
      await tester.pumpAndSettle(const Duration(seconds: 5));

      await tester.tap(find.byKey(const ValueKey('startButton')));
      await tester.pumpAndSettle();
      expect(find.text('Home Screen'), findsOneWidget);
      expect(find.text('Transactions'), findsOneWidget);
    });

    testWidgets('check value increments and data', (tester) async {
      app.main();
      await tester.pumpAndSettle(const Duration(seconds: 5));

      expect(find.text('Home Screen'), findsOneWidget);
      expect(find.text('Transactions'), findsOneWidget);

      await tester.tap(find.byTooltip('settingsTab'));
      await tester.pumpAndSettle();
    });
  });
}
```

Any helps?

**Comment by KazuCocoa** [comment]
I think you're asking general Flutter test framework's stuff rather than Appium.

https://api.flutter.dev/flutter/package-integration_test_integration_test/IntegrationTestWidgetsFlutterBinding/ensureInitialized.html

Please ask Flutter community's place instead of Appium's one

**Reply by jagadeeshpurplle** [reply]
Ohh sorry, it was mistaken
